src/energyplus/ForwardTranslator/ForwardTranslateDesignSpecificationOutdoorAir.cpp

```diff
@@ -22,43 +29,106 @@ namespace openstudio {
-      idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowperZone, flowPerZone);
-      idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowAirChangesperHour, ach);
+        idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowperPerson, flowPerPerson);
+        idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowperZoneFloorArea, flowPerArea);
+        idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowperZone, flowPerZone);
+        idfObject.setDouble(DesignSpecification_OutdoorAirFields::OutdoorAirFlowAirChangesperHour, ach);
 
-    } else {
-      LOG(Error, "Unknown OutdoorAirMethod '" << outdoorAirMethod << "' specified for OS:DesignSpecification:OutdoorAir named '"
+      } else {
+        LOG_FREE(Error, "openstudio.energyplus.ForwardTranslator",
+                 "Unknown OutdoorAirMethod '" << outdoorAirMethod << "' specified for OS:DesignSpecification:OutdoorAir named '"
                                               << modelObject.name().get() << "'");
+        return boost::none;
+      }
+
+      boost::optional<Schedule> schedule = modelObject.outdoorAirFlowRateFractionSchedule();
+      if (schedule) {
+        idfObject.setString(DesignSpecification_OutdoorAirFields::OutdoorAirScheduleName, schedule->name().get());
+      }
+
+      return idfObject;
+    }
+  }  // namespace detail
+
+  boost::optional<IdfObject> ForwardTranslator::getOrCreateThermalZoneDSOA(const model::ThermalZone& z) {
+
+    auto objInMapIt = m_zoneDSOAsMap.find(z.handle());
+    if (objInMapIt != m_zoneDSOAsMap.end()) {
+      return objInMapIt->second;
+    }
+
+    auto translateAndMapDSOA = [this](DesignSpecificationOutdoorAir& dsoa) -> boost::optional<IdfObject> {
+      auto objInMapIt = m_map.find(dsoa.handle());
+      if (objInMapIt != m_map.end()) {
+        return objInMapIt->second;
+      }
+
+      auto idf_dsoa_ = detail::translateDesignSpecificationOutdoorAir(dsoa);
+
+      if (idf_dsoa_) {
+        m_idfObjects.push_back(*idf_dsoa_);
+        m_map.emplace(dsoa.handle(), *idf_dsoa_);
+
+        if (m_progressBar) {
+          m_progressBar->setValue((int)m_map.size());
+        }
+      }
+      return idf_dsoa_;
+    };
+
+    auto spaces = z.spacesWithDesignSpecificationOutdoorAir();
+    if (spaces.empty()) {
+      m_zoneDSOAsMap.emplace(z.handle(), boost::none);
       return boost::none;
     }
 
-    boost::optional<Schedule> schedule = modelObject.outdoorAirFlowRateFractionSchedule();
-    if (schedule) {
-      idfObject.setString(DesignSpecification_OutdoorAirFields::OutdoorAirScheduleName, schedule->name().get());
+    OptionalIdfObject result;
+
+    if (m_forwardTranslatorOptions.excludeSpaceTranslation() || spaces.size() == 1) {
+      // Spaces, and therefore DSOAs have been combined already
+      auto dsoa = spaces.front().designSpecificationOutdoorAir().get();
+      result = translateAndMapDSOA(dsoa);
```

Comment:
If we don't do space translations we KNOW there's is exactly one (because it's not empty, we returned early).

or if we do but there's only one...

Then just translate it as is. No need for a DSOA:SpaceList with one DSOA on it.